ta4j-core/src/main/java/org/ta4j/core/criteria/NumberOfPositionsCriterion.java

```diff
@@ -43,9 +66,13 @@ public Num calculate(BarSeries series, TradingRecord tradingRecord) {
         return series.numOf(tradingRecord.getPositionCount());
     }
 
-    /** The lower the criterion value, the better. */
+    /**
+     * If {@link #lessBetter} == false, then the lower the criterion value, the
+     * better, otherwise the higher the criterion value the better.
+     */
     @Override
     public boolean betterThan(Num criterionValue1, Num criterionValue2) {
-        return criterionValue1.isLessThan(criterionValue2);
+        return lessBetter ? criterionValue1.isLessThan(criterionValue2)
+                : criterionValue1.isGreaterThan(criterionValue2);
```

Comment:
What about using an `int` for the `lessBetter` variable (for instance 0 or -1) and comparing the criterion values with `num.compareTo(num2) < lessBetter`

Reply:
I do not know if it is better or not. I like to reduce the amount of if-else statements or ternary operators if possible. I think it is just a matter of taste and I am also ok with this approach